support-lib/wasm/Future_wasm.hpp

```diff
@@ -34,28 +34,42 @@ class FutureAdaptor
 
     using NativePromiseType = Promise<CppResType>;
 
-    static void resolveNativePromise(int context, em::val res, em::val err) {
+    static void resolveNativePromise(int context, em::val res) {
         auto* pNativePromise = reinterpret_cast<NativePromiseType*>(context);
-        if (err.isNull() || err.isUndefined()) {
-            if constexpr (std::is_void_v<CppResType>) {
-                pNativePromise->setValue();
-            } else {
-                pNativePromise->setValue(RESULT::Boxed::toCpp(res));
-            }
+
+        if constexpr (std::is_void_v<CppResType>) {
+            pNativePromise->setValue();
         } else {
+            pNativePromise->setValue(RESULT::Boxed::toCpp(res));
+        }
+
+        delete pNativePromise;
+    }
+
+    static void rejectNativePromise(int context, em::val err) {
```

Comment:
Splitting up the rejection path. We overcomplicated this by having two params in one. It creates a lot more edge cases than dealing with the single parameter that the promise was resolved or rejected with.